public/app/features/logs/components/logParser.test.ts

```diff
@@ -6,6 +6,40 @@ import { getAllFields, createLogLineLinks, FieldDef } from './logParser';
 
 describe('logParser', () => {
   describe('getAllFields', () => {
+    it('should filter out fields with data links that have a nullish value', () => {
+      const createScenario = (value: unknown) =>
+        createLogRow({
+          entryFieldIndex: 1,
+          rowIndex: 0,
+          dataFrame: {
+            refId: 'A',
+            fields: [
+              testTimeField,
+              testLineField,
+              {
+                name: 'link',
+                type: FieldType.string,
+                config: {
+                  links: [
+                    {
+                      title: 'link1',
+                      url: 'https://example.com',
+                    },
+                  ],
+                },
+                values: [value],
+              },
+            ],
+            length: 1,
+          },
+        });
+
+      expect(getAllFields(createScenario(null))).toHaveLength(0);
+      expect(getAllFields(createScenario(undefined))).toHaveLength(0);
+      expect(getAllFields(createScenario(''))).toHaveLength(1);
+      expect(getAllFields(createScenario('test'))).toHaveLength(1);
```

Comment:
Maybe also `expect(getAllFields(createScenario(0))).toHaveLength(1);`?

Reply:
yeah, good point, i was also thinking about adding it, but technically, this is a `FieldType.string`... so i'm not sure.. still, if you also mention it, i'll add it (also `false` i guess 😄 )

Reply:
done